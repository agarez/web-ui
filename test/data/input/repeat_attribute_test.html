<!DOCTYPE html>
<!--
Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  <!-- Tests that repeat and template attribute work. -->
  <table>
    <tbody id='test'>
      <tr template repeat="row in table">
        <td template repeat="cell in row.where((c) => c != 0)">{{cell}}</td>
      </tr>
    </tbody>
  </table>
  <script type="application/dart">
  import 'dart:async';
  import 'dart:html';
  import 'package:unittest/unittest.dart';
  import 'package:web_ui/observe.dart';

  @observable
  List<List> table = toObservable([
    toObservable([1, 2, 3]),
    toObservable([4, 0, 5]),
    toObservable([0, 2, 4])
  ]);

  main() {
    Timer.run(() {
      table[1][1] = 9;
      Timer.run(() {
        var test = document.query('#test');
        expect(test.children.length, table.length + 1);
        expect(test.children[0].style.display, 'none');
        for (int row = 0; row < table.length; row++) {
          var tr = test.children[row + 1];
          expect(tr.tagName, 'TR');

          var filtered = table[row].where((c) => c != 0).toList();
          expect(tr.children.length, filtered.length + 1);
          expect(tr.children[0].style.display, 'none');

          for (int col = 0; col < filtered.length; col++) {
            var td = tr.children[col + 1];
            expect(td.tagName, 'TD');
            expect(td.innerHtml, filtered[col].toString());
          }
        }
        window.postMessage('done', '*');
      });
    });
  }
  </script>
</body>
</html>
