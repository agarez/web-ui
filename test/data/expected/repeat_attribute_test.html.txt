Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x84
  RenderBlock {HTML} at (0,0) size 800x84
    RenderBody {BODY} at (8,8) size 784x68
      RenderTable {TABLE} at (0,0) size 44x68
        RenderTableSection {TBODY} at (0,0) size 44x68
          RenderTableRow {TR} at (0,2) size 44x20
            RenderTableCell {TD} at (2,2) size 12x20 [r=0 c=0 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "1"
            RenderTableCell {TD} at (16,2) size 12x20 [r=0 c=1 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "2"
            RenderTableCell {TD} at (30,2) size 12x20 [r=0 c=2 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "3"
          RenderTableRow {TR} at (0,24) size 44x20
            RenderTableCell {TD} at (2,24) size 12x20 [r=1 c=0 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "4"
            RenderTableCell {TD} at (16,24) size 12x20 [r=1 c=1 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "9"
            RenderTableCell {TD} at (30,24) size 12x20 [r=1 c=2 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "5"
          RenderTableRow {TR} at (0,46) size 44x20
            RenderTableCell {TD} at (2,46) size 12x20 [r=2 c=0 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "2"
            RenderTableCell {TD} at (16,46) size 12x20 [r=2 c=1 rs=1 cs=1]
              RenderText {#text} at (1,1) size 10x18
                text run at (1,1) width 10: "4"
